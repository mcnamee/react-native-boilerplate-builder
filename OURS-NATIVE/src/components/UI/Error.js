import React from 'react';
import PropTypes from 'prop-types';
import {
  Container, Text, H3, Button, View,
} from 'native-base';
import Spacer from './Spacer';

const Error = ({ title, content, tryAgain }) => (
  <Container style={{ flex: 1 }}>
    <View style={{ alignSelf: 'center' }}>
      <Spacer size={20} />
      <H3 style={{ textAlign: 'center' }}>{title}</H3>
      <Text style={{ textAlign: 'center', marginBottom: 20 }}>{content}</Text>
      {tryAgain && (
        <Button block onPress={tryAgain}>
          <Text>Try Again</Text>
        </Button>
      )}
      <Spacer size={20} />
    </View>
  </Container>
);

Error.propTypes = {
  title: PropTypes.string,
  content: PropTypes.string,
  tryAgain: PropTypes.func,
};

Error.defaultProps = {
  title: 'Uh oh',
  content: 'An unexpected error came up',
  tryAgain: null,
};

export default Error;
