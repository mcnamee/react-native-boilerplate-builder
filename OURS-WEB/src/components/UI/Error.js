import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'reactstrap';
// import { Link } from 'react-router-dom';

const Error = ({ title, content }) => (
  <Row className="py-5 my-5">
    <Col md={{ size: 4, offset: 4 }}>
      <h2 className="text-center">{title}</h2>
      <p className="text-center">{content}</p>
      {/* <p className="text-center"><Link to="/" className="btn btn-primary">Go Home</Link></p> */}
    </Col>
  </Row>
);

Error.propTypes = {
  title: PropTypes.string,
  content: PropTypes.string,
};

Error.defaultProps = {
  title: 'Uh oh',
  content: 'An unexpected error came up',
};

export default Error;
