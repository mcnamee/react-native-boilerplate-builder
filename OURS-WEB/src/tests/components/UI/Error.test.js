import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Error from '../../../components/UI/Error';

it('<Error /> renders correctly', () => {
  const Component = <Error title="Hello" content="World" />;

  // Matches snapshot
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();

  // Has the correct text on the page
  const { getByText } = render(Component);
  expect(getByText('Hello')).toBeInTheDocument();
  expect(getByText('World')).toBeInTheDocument();
});
