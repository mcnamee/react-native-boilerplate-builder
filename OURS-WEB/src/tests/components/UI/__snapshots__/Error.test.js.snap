// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> renders correctly 1`] = `
<div
  className="py-5 my-5 row"
>
  <div
    className="col-md-4 offset-md-4"
  >
    <h2
      className="text-center"
    >
      Hello
    </h2>
    <p
      className="text-center"
    >
      World
    </p>
  </div>
</div>
`;
